avaframe/com2AB/com2AB.py

```diff
@@ -251,51 +280,58 @@ def readSplitPoint(fname, header):
     defname = 'SHP'
     SplitPoint = shpConv.SHP2Array(fname, defname)
-    splitPoint = SplitPoint['Coord']
-    if np.shape(splitPoint)[0] < 2:
+    splitPointx = SplitPoint['x']
+    splitPointy = SplitPoint['y']
+    if len(splitPointx) < 2:
         raise ValueError('split point file should contain at least 2 columns')
-    for i in range(np.shape(splitPoint)[1]):
-        Lx = int(np.round((splitPoint[0, i] - header.xllcorner) / header.cellsize))
-        Ly = int(np.round((splitPoint[1, i] - header.yllcorner) / header.cellsize))
+    for i in range(len(splitPointx)):
+        Lx = int(np.floor((splitPointx[i] - header.xllcorner) /
+                          header.cellsize))
+        Ly = int(np.floor((splitPointy[i] - header.yllcorner) /
+                          header.cellsize))
         if (Ly < 0 or Ly > header.nrows or Lx < 0 or Lx > header.ncols):
             raise ValueError('Split point is not on the DEM')
     return SplitPoint
 
 
-def checkProfile(indSplit, AvaProfile):
+def checkProfile(SplitPoint, AvaProfile):
     """ check that the avalanche profiles goes from top to bottom """
-    if AvaProfile[2, -1] > AvaProfile[2, 0]:
+    indSplit = SplitPoint['indSplit']
+    if AvaProfile['z'][-1] > AvaProfile['z'][0]:
         log.info('Profile reversed')
-        L = AvaProfile[3, -1]
-        AvaProfile = np.fliplr(AvaProfile)
-        AvaProfile[3, :] = L - AvaProfile[3, :]
-        indSplit = np.shape(AvaProfile)[1] - indSplit
-        return indSplit, AvaProfile
-    else:
-        return indSplit, AvaProfile
+        L = AvaProfile['s'][-1]
+        AvaProfile['x'] = np.flip(AvaProfile['x'])
+        AvaProfile['y'] = np.flip(AvaProfile['y'])
+        AvaProfile['z'] = np.flip(AvaProfile['z'])
+        AvaProfile['s'] = L - np.flip(AvaProfile['s'])
+        indSplit = len(AvaProfile['x']) - indSplit
 
+    return SplitPoint, AvaProfile
 
-def calcAB(eqInput, eqParameters):
-    """ Calculate Alpha Beta for data in eqInput according to chosen eqParameters """
+
+def calcAB(AvaProfile, eqParameters):
+    """
+    Calculate Alpha Beta for data in eqInput according to chosen eqParameters
+    """
     log.info("Calculating alpha beta")
     k1 = eqParameters['k1']
     k2 = eqParameters['k2']
     k3 = eqParameters['k3']
     k4 = eqParameters['k4']
     SD = eqParameters['SD']
 
-    s = eqInput['s']
-    x = eqInput['x']
-    y = eqInput['y']
-    z = eqInput['z']
-    indSplit = eqInput['indSplit']
-    eqOutput = eqInput.copy()
+    s = AvaProfile['s']
+    z = AvaProfile['z']
+    indSplit = AvaProfile['indSplit']
     ds = np.abs(s - np.roll(s, 1))
     dz = np.abs(z - np.roll(z, 1))
     ds[0] = 0.0
     dz[0] = 0.0
     angle = np.rad2deg(np.arctan2(dz, ds))
     CuSplit = s[indSplit]
+    # plt.figure(figsize=(10, 6))
+    # plt.plot(s, angle)
+    # plt.show()
     # TODO SPLIT POINT READING
     # get all values where Angle < 10 but >0
     # get index of first occurance and go one back to get previous value

```

Comment:
I am pretty confident it works well to remove all irregularities.
I left the debug blot because I would like to have a look at it on a real topography and see what it gives. Then the three points check would enable us to remove irregularities due to the real topography.

Reply:
hen make a flag inside the file itself (at the top) debugPlot = False, uncomment the code and put it in a if debugPlot:  construct ... (-> thats a FeatureFlag I was talking about at one point...)

This is for things which we need for now, but will get of rid in future...